File: Step 1 - Learn the Basics/1.4 Know Basic Maths/01 Count Digits/CountDigits.py
# https://www.geeksforgeeks.org/problems/count-digits5716/1

n = int(input("Enter a number: "))
temp = n
count = 0

while n > 0:
    rem = n % 10
    if rem != 0 and temp % rem == 0:
        count += 1
    n //= 10

print(f"The count of digits that evenly divide {temp} is: {count}")

File: Step 1 - Learn the Basics/1.4 Know Basic Maths/01 Count Digits/CountDigits.java
// https://www.geeksforgeeks.org/problems/count-digits5716/1

public class CountDigits {
    static int evenlyDivides(int n) {
        // code here
        int temp = n;
        int rem = 0;
        int count =0;
        while(n>0){
            rem = n%10;
            if(rem!=0 && temp%rem==0)
                count++;
            n/=10;
        }
        return count;
    }
}
